src/decode.c

```diff
@@ -419,7 +425,14 @@ void DecodeRegisterPerfCounters(DecodeThreadVars *dtv, ThreadVars *tv)
         StatsRegisterCounter("defrag.ipv6.timeouts", tv);
     dtv->counter_defrag_max_hit =
         StatsRegisterCounter("defrag.max_frag_hits", tv);
-
+    
+    int i = 0;
+    for (i = 0; i < DECODE_EVENT_PACKET_MAX; i++) {
+        char counter_name[256];
+        snprintf(counter_name, 256, "decoder.%s", DEvents[i].event_name);
```

Comment:
I don't like how this makes dynamically allocated counter names necessary again (had just removed that recently).

I'm thinking that we should sync the decoder and stream names. That would mean we prepend 'decoder.' to each decoder name string so we can use those directly in the counter.

It would require an update to DetectDecodeEventSetup, similar to how DetectStreamEventSetup handles it.